dockers/tpu-tests/tpu_test_cases.jsonnet

```
@@ -10,17 +10,31 @@ local tputests = base.BaseTest {
 
   timeout: 900, # 15 minutes, in seconds.
 
-  image: std.extVar('image'),
-  imageTag: std.extVar('image-tag'),
+  image: 'pytorchlightning/pytorch_lightning:base-xla-py3.6-torch{PYTORCH_VERSION}',
+  imageTag: '{PR_NUMBER}-{PYTORCH_VERSION}',
 
   tpuSettings+: {
-    softwareVersion: 'pytorch-VERSION',
+    softwareVersion: 'pytorch-{PYTORCH_VERSION}',
   },
   accelerator: tpus.v3_8,
 
   command: utils.scriptCommand(
     |||
+      mkdir -p /home/runner/work/pytorch-lightning && cd /home/runner/work/pytorch-lightning
+      git clone https://github.com/PyTorchLightning/pytorch-lightning.git
       cd pytorch-lightning
+      echo $PWD
+      git ls-remote --refs origin
+      git fetch origin "refs/pull/{PR_NUMBER}/head:pr/{PR_NUMBER}" && git checkout "pr/{PR_NUMBER}"
+
+      # drop horovod as it is not needed
+      python -c "fname = 'pytorch-lightning/requirements/extra.txt' ; lines = [line for line in open(fname).readlines() if not line.startswith('horovod')] ; open(fname, 'w').writelines(lines)"
+      # drop fairscale as it is not needed
+      python -c "fname = 'pytorch-lightning/requirements/extra.txt' ; lines = [line for line in open(fname).readlines() if 'fairscale' not in line] ; open(fname, 'w').writelines(lines)"
+
+      pip install -e .
```

```suggestion
      pip install -e ".[dev]"
```
but why install itself?

---

We need to re-install so we test the commit changes.